gce/prepull-1.18.yaml

```diff
@@ -0,0 +1,161 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: prepull-test-containers
+spec:
+  selector:
+    matchLabels:
+      prepull-test-images: e2e
+  template:
+    metadata:
+      labels:
+        prepull-test-images: e2e
+    spec:
+      nodeSelector:
+        kubernetes.io/os: windows
+      # To generate this list from the build-log.txt output of an e2e test
+      # result:
+      #   grep -o -E 'e2eteam/([[:alnum:]_-]+):([[:alnum:]./_-]+)' \
+      #   build-log.txt | sort | uniq
+      # DaemonSets do not support a RestartPolicy other than 'Always', so we
+      # run ping in each container to keep it alive so that kubernetes does not
+      # continually restart the containers while we're prepulling.
+      containers:
+      - image: e2eteam/agnhost:2.8
+        name: agnhost-28
+        resources:
+          requests:
+            cpu: 1m
+        command: ['cmd.exe', '/c', 'ping -n 1800 127.0.0.1 >NUL']
+      - image: e2eteam/busybox:1.29
+        name: busybox1
+        resources:
+          requests:
+            cpu: 1m
+        command: ['cmd.exe', '/c', 'ping -n 1800 127.0.0.1 >NUL']
+      - image: e2eteam/cassandra:v13
+        name: cassandra5
+        resources:
+          requests:
+            cpu: 1m
+        command: ['cmd.exe', '/c', 'ping -n 1800 127.0.0.1 >NUL']
+      - image: e2eteam/curl:1803
+        name: curl2
+        resources:
+          requests:
+            cpu: 1m
+        command: ['cmd.exe', '/c', 'ping -n 1800 127.0.0.1 >NUL']
+      - image: e2eteam/dnsutils:1.1
+        name: dnsutils6
+        resources:
+          requests:
+            cpu: 1m
+        command: ['cmd.exe', '/c', 'ping -n 1800 127.0.0.1 >NUL']
+      - image: e2eteam/echoserver:2.2
+        name: echoserver7
+        resources:
+          requests:
+            cpu: 1m
+        command: ['cmd.exe', '/c', 'ping -n 1800 127.0.0.1 >NUL']
+      - image: e2eteam/entrypoint-tester:1.0
+        name: entrypoint-tester8
+        resources:
+          requests:
+            cpu: 1m
+        command: ['cmd.exe', '/c', 'ping -n 1800 127.0.0.1 >NUL']
+      - image: e2eteam/etcd:3.3.10
+        name: etcd9
+        resources:
+          requests:
+            cpu: 1m
+        command: ['cmd.exe', '/c', 'ping -n 1800 127.0.0.1 >NUL']
+      - image: e2eteam/etcd:3.4.3
+        name: etcd-343
+        resources:
+          requests:
+            cpu: 1m
+        command: ['cmd.exe', '/c', 'ping -n 1800 127.0.0.1 >NUL']
+      - image: e2eteam/etcd:v3.3.10
```

Comment:
it is a dup? two etcd:v3.3.10?

why both 3.3.10 and 3.4.3 are needed?

Reply:
Here's a 2019 master test run that appears to use both `e2eteam/etcd:3.3.10` and `e2eteam/etcd:v3.3.10`: https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-windows-gce-2019/1227005233964519425.

Here's a 1909 master test run that appears to use just `e2eteam/etcd:3.4.3`: https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-windows-gce-1909/1227006493195243520

I'm not immediately sure why the containers are different... also, my regex that's parsing the test logs may not be perfect. I'll dig a little deeper...

Reply:
Sigh... it looks like the list of containers that I'm pulling out of `build-log.txt` from current test runs may be a subset of the containers that are actually used. I'm comparing test logs from today to test logs from last week for each k8s version and they're different. Ugh.

I don't really know why a container might be named in the test output from one test run but not another. I didn't spend a lot of time looking at the context surrounding the container names in the test log files.

Since we rejected the approach of using a prepull manifest with every-version-of-every-container I don't know of a much better approach than parsing the test log. One idea I have now is to run a `docker image list` command on each node at the end of the test run to see what images have been pulled. Of course this might fail to capture any containers that failed to pull in time during the test run, but by combining this with the regex parsing approach and looking at a few different test runs hopefully we can approach an accurate set of test containers.

Ugh.

Reply:
Just 3.4.3 is needed.

Reply:
Ok. As noted above I see 3.3.10 in some of the test logs, but on closer inspection it's not clear that 3.3.10 is used in any tests - it's probably just appearing in the log *because of* the prepull we're already doing.

Reply:
Done in https://github.com/kubernetes-sigs/windows-testing/pull/130.